test/functional/test_framework/mininode.py

```diff
@@ -440,3 +454,127 @@ def network_thread_join(timeout=10):
     for thread in network_threads:
         thread.join(timeout)
         assert not thread.is_alive()
+
+class P2PStub(P2PInterface):
+    """A P2P Stub class.
+
+    Keeps a block and transaction store and responds correctly to getdata and getheaders requests."""
+
+    def __init__(self):
+        super().__init__()
+        self.reject_code_received = None
+        self.reject_reason_received = None
+        self.block_store = OrderedDict()
+        self.tx_store = {}
+        self.getdata_requests = []
+
+    def on_getdata(self, message):
+        """Check for the tx/block in our stores and if found, reply with an inv message."""
+        for inv in message.inv:
+            self.getdata_requests.append(inv.hash)
+            if inv.type == 1 or inv.type == 1 | MSG_WITNESS_FLAG:
+                if inv.hash in self.tx_store.keys():
+                    self.send_message(msg_tx(self.tx_store[inv.hash]))
+            elif inv.type == 2 or inv.type == 2 | MSG_WITNESS_FLAG:
+                if inv.hash in self.block_store.keys():
+                    self.send_message(msg_block(self.block_store[inv.hash]))
+
+    def on_getheaders(self, message):
+        """Search back through our block store for the locator, and reply with a headers message if found."""
+
+        locator, hash_stop = message.locator, message.hashstop
+
+        # Assume that the most recent block added is the tip
+        if len(self.block_store) == 0:
+            return
+        current_block_header = self.block_store[next(reversed(self.block_store))]
+        if current_block_header is None:
```

Comment:
nit: Add a comment why this would ever be true and not dead code?

Reply:
This isn't actually required. Removing.